src/api.ts

```diff
@@ -298,11 +294,12 @@ router.post(
       return res.status(400).send(validateBidCancelSchema.errors);
     }
     try {
-      const bidData: Bid | null = await database.getBidBySignedMessage(
+      const bidData: CancelableBid | null = await database.getBidBySignedMessage(
         req.body.bidMessageSignature
       );
 
       if (!bidData) return res.status(400).send("Bid not found");
+      if (bidData.cancelDate && bidData.cancelDate > 0) return res.status(404).send("Bid is no longer active");
```

Comment:
This should be a 400, not a 404. A 404 tells me that the resource at `bid/cancel/{data}` was not found. In this case, it was found, but the bid is no longer active, and thus an invalid request.